src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8610,7 +8610,7 @@ static bool tryResolveExtensionInScope(
                     lookupResult, analyzedArguments, ref actualMethodArguments, options, in callingConvention, binder, diagnostics);
 
                 // 3. resolve properties
-                OverloadResolutionResult<PropertySymbol>? propertyResult = resolveProperties(left, lookupResult, binder, ref actualReceiverArguments, ref useSiteInfo);
+                OverloadResolutionResult<PropertySymbol>? propertyResult = arity != 0 ? null : resolveProperties(left, lookupResult, binder, ref actualReceiverArguments, ref useSiteInfo);
```

Comment:
>arity != 0 ? null :  [](http://example.com/codeflow?start=75&length=20)

Given a comment in ```LookupExtensionMembersInSingleBinder```, I think this logic should not be needed for correctness, there should be no properties in the lookup result and this code is merely an optimization. Consider asserting that.